src/osd/PGBackend.cc

```diff
@@ -923,13 +991,9 @@ map<pg_shard_t, ScrubMap *>::const_iterator
     }
 
 out:
-    // Check error_string because some errors already generated messages
-    if (error_string != "") {
-      dout(10) << __func__ << ": error(s) osd " << j->first
-	       << " for obj " << obj
-	       << "," << error_string
-	       << dendl;
-    }
+    if (error)
+        errorstream << pgid.pgid << " shard " << l << ": soid " << obj
```

Comment:
nit, probably we can be more consistent in the error message format? if we have an object level scrub error:
```c++
      errorstream << pgid.pgid << " soid " << *k
		  << ": failed to pick suitable object info\n";
```

and a shard error looks like
```
<pgid> shard 1: soid 3:ce3f1d6a:::ROBJ1:head size 9 != size 7 from auth oi
```
in which, the detailed error is started with a ":".  and the object id is placed right next to pgid. i think it makes more sense in this context, as it'd easier to identify the error in the line . and if we put the object id next to <pgid>, will it help the user to sort the output?